src/formatting/items.rs

```diff
@@ -854,12 +866,25 @@ pub(crate) fn format_impl(
             visitor.block_indent = item_indent;
             visitor.last_pos = lo + BytePos(open_pos as u32);
 
+            let open_nls = &visitor.trim_spaces_after_opening_brace(
+                inner_attributes(&item.attrs)
+                    .first()
+                    .map(|a| a.span.lo())
+                    .or(items.first().map(|i| i.span().lo())),
```

Comment:
Bit of a nit but could we use `map_or` here or even `map_or_else` to get the lazy evaluation? 

Something like this perhaps (usual disclaimer about inline GH suggestions not being formatted correctly)
```suggestion
                inner_attributes(&item.attrs).first().map_or_else(
                    || items.first().map(|i| i.span().lo()),
                    |a| Some(a.span.lo()),
                ),
```

Reply:
Done